projects/core/src/error-handling/http-error-handler/http-error-handler.interceptor.spec.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  HttpErrorResponse,
+  HttpEvent,
+  HttpHandler,
+  HttpRequest,
+} from '@angular/common/http';
+import { Observable, throwError } from 'rxjs';
+import { TestBed } from '@angular/core/testing';
+import { HttpErrorHandlerInterceptor } from './http-error-handler.interceptor';
+import { ErrorHandler, Injectable } from '@angular/core';
+
+@Injectable()
+class MockErrorHandler {
+  handleError(_error: any): void {}
+}
+
+describe('HttpErrorHandlerInterceptor', () => {
+  let interceptor: HttpErrorHandlerInterceptor;
+  let errorHandler: ErrorHandler;
+  let request: HttpRequest<any>;
+  let next: HttpHandler;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        HttpErrorHandlerInterceptor,
+        { provide: ErrorHandler, useClass: MockErrorHandler },
+      ],
+    });
+
+    interceptor = TestBed.inject(HttpErrorHandlerInterceptor);
+    errorHandler = TestBed.inject(ErrorHandler);
+
+    request = new HttpRequest('GET', 'test-url');
+    next = {
+      handle: () => new Observable<HttpEvent<any>>(),
+    } as HttpHandler;
+  });
+
+  it('should create the interceptor', () => {
+    expect(interceptor).toBeTruthy();
+  });
+
+  it('should call handleError on error', () => {
+    const error: HttpErrorResponse = new HttpErrorResponse({
+      status: 400,
+      statusText: 'error',
+    });
+    spyOn(errorHandler, 'handleError');
+
+    next.handle = () => throwError(error);
+
+    interceptor.intercept(request, next).subscribe({
+      error: (err) => {
+        expect(err).toEqual(error);
+        expect(errorHandler.handleError).toHaveBeenCalledWith(error);
+      },
```

Comment:
Let's include the Jasmine `done` callback inside the body of the `subscribe()` method. Otherwise, there is a possibility that the callback function may not be executed at all. This can happen if the observable doesn't emit any values due to a bug in the code-under-test or a mistake in the mocking logic.

(the same remark applies to subscription in line 69-71).

Reply:
fixed